src/osd/Watch.cc

```diff
@@ -393,7 +393,13 @@ void Watch::connect(ConnectionRef con, bool _will_ping)
     last_ping = ceph_clock_now();
     register_cb();
   } else {
-    unregister_cb();
+    if (!con->get_priv()) {
+      // if session is already nullptr
+      // !will_ping should also register WatchTimeout
+      register_cb();
```

Comment:
Should we also reset `conn` with `conn = ConnectionRef()` before registering the timeout, to fully mimic the connection reset handler?

Reply:
> Should we also reset `conn` with `conn = ConnectionRef()` before registering the timeout, to fully mimic the connection reset handler?

Means `conn` pointer in `class watch` will remain invalid? 
At first writing, I actually wanted to invole `Watch::disconnect` directly instread of `Watch::register_cb`. However it looks not appropriate to invole `Watch::disconnect` in `Watch::connect`.

Reply:
done. thx